parquet-hadoop/src/test/java/org/apache/parquet/statistics/TestFloat16Statistics.java

```diff
@@ -0,0 +1,272 @@
+
+  private short[] valuesInAscendingOrder = {
+    (short) 0xfc00, // -Infinity
+    (short) 0xc000, // -2.0
+    -Float16.MAX_VALUE, // -6.109476E-5
+    Float16.NEGATIVE_ZERO, // -0
+    Float16.POSITIVE_ZERO, // +0
+    Float16.MIN_VALUE, // 5.9604645E-8
+    Float16.MAX_VALUE, // 65504.0
+    (short) 0x7c00}; // Infinity
+
+  private short[] valuesInAscendingOrderMinMax = {
+    (short) 0xfc00, // -Infinity
+    (short) 0x7c00}; // Infinity
+
+  private short[] valuesInDescendingOrder = {
+    (short) 0x7c00, // Infinity
+    Float16.MAX_VALUE, // 65504.0
+    Float16.MIN_VALUE, // 5.9604645E-8
+    Float16.POSITIVE_ZERO, // +0
+    Float16.NEGATIVE_ZERO, // -0
+    -Float16.MAX_VALUE, // -6.109476E-5
+    (short) 0xc000, // -2.0
+    (short) 0xfc00}; // -Infinity
+
+  private short[] valuesInDescendingOrderMinMax = {
+    (short) 0xfc00, // -Infinity
+    (short) 0x7c00}; // Infinity
+
+  private short[] valuesUndefinedOrder = {
+    Float16.MAX_VALUE, // 65504.0
+    (short) 0x7c00, // Infinity
+    Float16.NEGATIVE_ZERO, // -0
+    Float16.MIN_VALUE, // 5.9604645E-8
+    Float16.POSITIVE_ZERO, // +0
+    (short) 0xc000, // -2.0
+    -Float16.MAX_VALUE, // -6.109476E-5
+    (short) 0xfc00}; // -Infinity
+
+  private short[] valuesUndefinedOrderMinMax = {
+    (short) 0xfc00, // -Infinity
+    (short) 0x7c00}; // Infinity
+
+  private short[] valuesAllPositiveZero = {
+    Float16.POSITIVE_ZERO, // +0
+    Float16.POSITIVE_ZERO, // +0
+    Float16.POSITIVE_ZERO, // +0
+    Float16.POSITIVE_ZERO}; // +0
+
+  private short[] valuesAllPositiveZeroMinMax = {
+    Float16.POSITIVE_ZERO, // +0
+    Float16.POSITIVE_ZERO}; // +0
+
+  // Float16Statistics: Updating min to -0.0 to ensure that no 0.0 values would be skipped
+  private short[] valuesAllPositiveStatsZeroMinMax = {
+    Float16.NEGATIVE_ZERO, // -0
+    Float16.POSITIVE_ZERO}; // +0
+
+  private short[] valuesAllNegativeZero = {
+    Float16.NEGATIVE_ZERO, // -0
+    Float16.NEGATIVE_ZERO, // -0
+    Float16.NEGATIVE_ZERO, // -0
+    Float16.NEGATIVE_ZERO}; // -0
+
+  private short[] valuesAllNegativeZeroMinMax = {
+    Float16.NEGATIVE_ZERO, // -0
+    Float16.NEGATIVE_ZERO}; // -0
+
+  // Float16Statistics: Updating max to +0.0 to ensure that no 0.0 values would be skipped
+  private short[] valuesAllNegativeStatsZeroMinMax = {
+    Float16.NEGATIVE_ZERO, // -0
+    Float16.POSITIVE_ZERO}; // +0
+
+  private short[] valuesWithNaN = {
+    (short) 0xc000, // -2.0
+    Float16.MAX_VALUE, // 65504.0
+    Float16.MIN_VALUE, // 5.9604645E-8
+    Float16.NaN}; // NaN
+
+  private short[] valuesWithNaNMinMax = {
+    (short) 0xc000, // -2.0
+    Float16.NaN}; // NaN
+
+  // Float16Statistics: Drop min/max values in case of NaN as the sorting order of values is undefined
+  private short[] valuesWithNaNStatsMinMax = {
+    Float16.POSITIVE_ZERO, // +0
+    Float16.POSITIVE_ZERO}; // +0
+
+  @Test
+  public void testFloat16ColumnIndex() throws IOException {
```

Comment:
It would be good to move this test and below to a new file called `TestFloat16ReadWriteRoundTrip.java`. So this file can solely test the Float16Statistics class.